lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -4655,6 +4679,7 @@ class ConvertTorchToTosa : public ConvertTorchToTosaBase<ConvertTorchToTosa> {
     INSERT_ATENOP_PATTERN(AtenToDtypeOp);
     INSERT_ATENOP_PATTERN(AtenConstantPadNdOp);
     INSERT_ATENOP_PATTERN(AtenRemainderScalarOp);
+    INSERT_ATENOP_PATTERN(AtenCatOp);
```

Comment:
Wondering why tests like [TensorsConcatModule](https://github.com/llvm/torch-mlir/blob/a13d301356fe8fe414a707ed3a3e011cd888e3bb/python/torch_mlir_e2e_test/test_suite/basic.py#L576) fail. If it must fail (I guess because the tensors have dynamic shapes) then you probably need to replicate the existing `cat` tests to make `Static` versions as you see elsewhere in `basic.py` for other operators.

Reply:
for tosa I could only get positive dim to work(https://mlir.llvm.org/docs/Dialects/TOSA/#tosaconcat-mlirtosaconcatop), so just have one static positive dim test. I have renamed the test as you suggested

Reply:
just add support for negative dim in the cpp side and using torch.cat now